src/Maths.hs

```diff
@@ -0,0 +1,50 @@
+module Maths
+  ( intersecPoint
+  , dot
+  ) where
+
+-- aliases
+type Vector = (Float, Float)
+type SquareMatrix = (Vector, Vector)
+type Point = (Float, Float)
+
+-- | Calculate the dot product
+dot :: Vector   -- ^ first vector called va
+    -> Vector   -- ^ second vector called vb
+    -> Float    -- ^ Scalar dot product
+dot (ax, ay) (bx, by) = (ax * bx) + (ay * by)
+
+-- | Calculate determinant of a vector
+determinant :: SquareMatrix
+            -> Float    -- ^ determinant
+determinant ((ax, ay), (bx, by)) = ax * by - bx * ay
+
+-- | Use the Cramer theorem to resolve a linear equation
+cramer :: SquareMatrix      -- ^ base (inversible matrix)
+       -> Vector            -- ^ vector
+       -> Vector            -- ^ vector with unknown variables resolved
+cramer ((v1x, v1y), (v2x, v2y)) (rx, ry) =
+      ( determinant ((rx, ry), (v2x, v2y)) / det
+      , determinant ((v1x, v1y), (rx, ry)) / det)
+      where
+        det = determinant ((v1x, v1y), (v2x, v2y))
+
+-- | return the intersection point between 2 vectors, if exists
```

Comment:
Not between two vectors: given two *segments*, each one being defined by a starting point and a vector.